doc/manual/src/glossary.md

```diff
@@ -217,6 +217,19 @@
 
   [output path]: #gloss-output-path
 
+- [deriving path]{#gloss-deriving-path}
+
+  A [store]-level expression (not Nix language expression) that denotes a [store object].
```

Comment:
```suggestion
  Deriving paths are a way to refer to [store objects][store object] that are not yet [realised][realise].
  This is necessary because, in general and particularly for [content-addressed derivations][content-addressed derivation], the [output path] of an [output] is not known in advance.
```

I really struggle with "store-level expression". This is just very confusing, even if one could argue there is exists some sort of "store language".